lib/elixir/lib/code.ex

```diff
@@ -1045,6 +1045,66 @@ defmodule Code do
   When given a path to a `.beam` file, it will load the docs directly from that
   file.
 
+  It returns the term stored in the documentation chunk in the format defined by
+  [EEP 48](http://erlang.org/eep/eeps/eep-0048.html) or `{:error, reason}` if
+  the chunk is not available.
+
+  ## Examples
+
+      # Module documentation of an existing module
+      iex> {:docs_v1, _, :elixir, _, %{"en" => module_doc}, _, _} = Code.fetch_docs(Atom)
+      iex> module_doc |> String.split("\n") |> Enum.at(0)
+      "Convenience functions for working with atoms."
+
+      # A module that doesn't exist
+      iex> Code.fetch_docs(ModuleNotGood)
+      {:error, :module_not_found}
+
+  """
+  @spec fetch_docs(module) ::
+          {:docs_v1, anno, beam_language, format, module_doc :: doc, metadata,
+           docs :: [{{kind, name, arity}, anno, signature, doc, metadata}]}
+          | {:error, reason :: term}
+        when anno: :erl_anno.anno(),
+             beam_language: atom,
+             format: binary,
+             doc: %{binary => binary} | :none | :hidden,
+             kind: atom,
+             name: atom,
+             signature: [binary],
+             metadata: map
```

Comment:
There might be a better way to write this spec. I'm not sure if I can name some parts of the spec for more clarity (e.g. `module_doc :: doc`).

Reply:
What do you think about being more explicit about the error responses `{:error, :module_not_found | : docs_chunk_not_found}` ?

Reply:
I think that's a reasonable thing to do as long as the list of possible errors are limited as it is in this case.